api/agent/agent.go

```diff
@@ -1239,29 +1227,30 @@ func newHotContainer(ctx context.Context, call *call, cfg *Config, id string, ud
 	}
 }
 
-func (c *container) swap(stdout, stderr io.Writer, cs *drivers.Stats) func() {
-	// if tests don't catch this, then fuck me
-	ostdout := c.stdout.(common.GhostWriter).Swap(stdout)
-	ostderr := c.stderr.(common.GhostWriter).Swap(stderr)
-
+func (c *container) swap(stderr io.Writer, cs *drivers.Stats) func() {
 	c.swapMu.Lock()
+	c.stderr = stderr
 	ocs := c.stats
 	c.stats = cs
 	c.swapMu.Unlock()
 
 	return func() {
-		c.stdout.(common.GhostWriter).Swap(ostdout)
-		c.stderr.(common.GhostWriter).Swap(ostderr)
 		c.swapMu.Lock()
+		c.stderr = stderr
```

Comment:
Is this supposed to put the old stderr back?

Reply:
that's a bingo